snapcraft/tests/fake_servers.py

```diff
@@ -0,0 +1,277 @@
+            logger.error(
+                'Not implemented path in fake Store Search server: {}'.format(
+                    self.path))
+            raise NotImplementedError(self.path)
+
+    def _handle_search_request(self, query):
+        if len(query) > 1 or 'package_name:' not in query[0]:
+            logger.error(
+                'Not implemented query in fake Store Search server: {}'.format(
+                    query))
+            raise NotImplementedError(query)
+        query = query[0]
+        package = query.split('package_name:')[1].strip('"')
+        logger.debug(
+            'Handling search request for package {}, with headers {}'.format(
+                package, self.headers))
+        self.send_response(200)
+        self.send_header('Content-Type', 'application/hal+json')
+        self.end_headers()
+        if package == 'test-snap':
+            response = {'_embedded': {
+                'clickindex:package': [
+                    {'download_url': urllib.parse.urljoin(
+                        'http://localhost:{}'.format(self.server.server_port),
+                        'download-snap/test-snap.snap'),
+                     # sha512sum snapcraft/tests/data/test-snap.snap
+                     'download_sha512': (
+                        '69d57dcacf4f126592d4e6ff689ad8bb8a083c7b9fe44f6e738ef'
+                        'd22a956457f14146f7f067b47bd976cf0292f2993ad864ccb498b'
+                        'fda4128234e4c201f28fe9')}]}}
+        else:
+            response = {}
+        self.wfile.write(json.dumps(response).encode())
+
+    def _handle_download_request(self, snap):
+        logger.debug('Handling download request for snap {}'.format(snap))
+        self.send_response(200)
+        self.send_header('Content-Type', 'application/octet-stream')
+        self.end_headers()
+        # TODO create a test snap during the test. --elopio - 2016-05-01
+        snap_path = os.path.join(
+            os.path.dirname(snapcraft.tests.__file__), 'data',
+            'test-snap.snap')
+        with open(snap_path, 'rb') as snap_file:
+            self.wfile.write(snap_file.read())
+
+
+class FakeStoreUploadServer(http.server.HTTPServer):
+
+    def __init__(self, server_address):
+        super().__init__(
+            server_address, FakeStoreUploadRequestHandler)
+
+
+class FakeStoreUploadRequestHandler(http.server.BaseHTTPRequestHandler):
+
+    def do_POST(self):
+        parsed_path = urllib.parse.urlparse(self.path)
+        if parsed_path.path.startswith('/unscanned-upload/'):
+            self._handle_upload_request()
+        else:
+            logger.error(
+                'Not implemented path in fake Store Search server: {}'.format(
+                    self.path))
+            raise NotImplementedError(self.path)
+
+    def _handle_upload_request(self):
+        logger.info('Handling upload request')
+        self.send_response(200)
+        self.send_header('Content-Type', 'application/octet-stream')
```

Comment:
Not sure about that one, shouldn't that be 'Content-Type': 'application/json' ? No test were failing with that ?